app/scripts/modules/core/src/presentation/tables/standardGridTableLayout.tsx

```diff
@@ -0,0 +1,139 @@
+    gridColumns.push(EXPAND_COLLAPSE_COLUMN_SIZE);
+  }
+
+  return gridColumns.join(' ');
+};
+
+const TableLayout = ({
+  columns,
+  expandable,
+  expandAll,
+  isMobile,
+  children,
+  sizes,
+}: ITableLayoutProps & { sizes: ITableColumnSizes }) => {
+  const gridTemplateColumns = getGridColumnsStyle(sizes, expandable, isMobile);
+
+  return (
+    <div className="StandardGridTableLayout">
+      {!isMobile && (
+        <div className="standard-grid-table-header" style={{ gridTemplateColumns }}>
+          {columns.map(({ name, sortDirection, onSort }) => (
+            <div
+              key={name}
+              className={classNames('standard-grid-table-cell', { sortable: !!onSort })}
+              onClick={onSort || null}
+            >
+              {name}
+              {sortDirection ? <span className={classNames('sort-indicator', sortDirection)} /> : null}
+            </div>
+          ))}
+          {expandable && (
+            <div className="standard-grid-table-cell header-expand-toggle" onClick={expandAll}>
+              <i className="ico icon-arrow-toggle-all" />
+            </div>
+          )}
+        </div>
+      )}
+      {children}
+    </div>
+  );
+};
+
+const TableRowLayout = ({
+  expandable,
+  expanded,
+  setExpanded,
+  isMobile,
+  sizes,
+  children,
+  renderExpandedContent,
+}: ITableRowLayoutProps & { sizes: ITableColumnSizes }) => {
+  const gridTemplateColumns = getGridColumnsStyle(sizes, expandable, isMobile);
+
+  return (
+    <>
+      <div
+        className={classNames('standard-grid-table-row', { expandable, expanded })}
+        style={{ gridTemplateColumns }}
+        onClick={expandable ? () => setExpanded(!expanded) : null}
+      >
+        {children}
+        {expandable && !isMobile && (
+          <div className="standard-grid-table-cell row-expand-toggle">
+            <i className={classNames('ico', { 'icon-arrow-collapsed': !expanded, 'icon-arrow-expanded': expanded })} />
+          </div>
+        )}
+      </div>
+      {expandable && expanded && <div className="expanded-row-content">{renderExpandedContent()}</div>}
+    </>
+  );
+};
+
+const TableCellLayout = ({ children, indent, index, columns, isMobile }: ITableCellLayoutProps) => {
+  const columnName = columns[index].name;
+  const indentPx = !!indent && isMobile ? `${GUTTER_SIZE_PX + indent * INDENT_SIZE_PX}px` : undefined;
+
+  return (
+    <div className="standard-grid-table-cell" style={{ paddingLeft: indentPx }}>
+      {isMobile && <div className="mobile-cell-header">{columnName}</div>}
+      <div className="cell-content">{children}</div>
+    </div>
+  );
+};
+
+export const standardGridTableLayout = (sizes: ITableColumnSizes) => ({
+  getTableLayout: (props: ITableLayoutProps) => <TableLayout {...props} sizes={sizes} />,
+  getTableRowLayout: (props: ITableRowLayoutProps) => <TableRowLayout {...props} sizes={sizes} />,
+  getTableCellLayout: (props: ITableCellLayoutProps) => <TableCellLayout {...props} />,
+});
```

Comment:
this is an example of how a layout can accept some of its own layout-specific configuration without jumping through major hoops.